invenio_communities/members/services/service.py

```diff
@@ -420,7 +459,7 @@ def _members_search(
             extra_filter=filter,
             **kwargs
         )
-        search_result = search.execute()
+        search_result = search.scan() if scan else search.execute()
```

Comment:
to consider, not necessarily applying here (how many community members we might have?)
the scan might be problematic without setting timeout for scroll 
(see usage in https://github.com/CERNDocumentServer/cds-ils/blob/215c8c75790a6493b1365c7d47633e9df1cd4eae/cds_ils/migrator/relations/series.py#L63) 
the scan timeouts after a while, see https://opensearch.org/docs/latest/api-reference/scroll/

Reply:
Indeed, it should have a sane default, overridable via args.

Reply:
There is a default timeout set to 5 minutes: https://github.com/opensearch-project/opensearch-py/blob/fe3b5a8922aa8eb04f735c74d127d7ea68a00bec/opensearchpy/helpers/actions.py#L492-L503

As I understand, the scroll timeout will also be reset after each "page" (all hits for the current scroll action have been exhausted): https://github.com/opensearch-project/opensearch-py/blob/fe3b5a8922aa8eb04f735c74d127d7ea68a00bec/opensearchpy/helpers/actions.py#L598-L600
As an example, calling scan will:
- perform a scroll search
- fetch 1000 hits (out of 2000)
- keep the state alive for 5 minutes
- yield hit for hit of current hits
- after exhausting the current hits, start again, until there are no more new hits available

It should also already be overridable via something like `search_preference={"scroll": "4h"}`. Though I have not tested this.

Do you think the default 5 minutes is too much? For OAI set percolators, we have set the default to 1 minute:
- https://github.com/inveniosoftware/invenio-oaiserver/blob/42ed11d1963ce2b0a6a18cec99b970e27c9a08ce/invenio_oaiserver/percolator.py#L152-L157
- https://github.com/inveniosoftware/invenio-oaiserver/blob/42ed11d1963ce2b0a6a18cec99b970e27c9a08ce/invenio_oaiserver/config.py#L55

Reply:
discussed IRL, agreed on adding a comment reminding about the default scroll time

Reply:
did a bit of digging and testing. seems like the scroll params are not overridable so easily (sorry about that info prior).

We could add explicit `scan_params` to the `_members_search` method, and depending on the scan flag, add these params to the search before starting the scan.

```py
    def _members_search(...,
        scan=False,
        # add explicit scan_params
        scan_params=None,
        **kwargs
    ):
        """Members search."""
        ...
        # Prepare and execute the search
        params = params or {}
        # set explicit scan_params
        scan_params = scan_params or {}

        search = self._search(
            "search_members",
            identity,
            params,
            search_preference,
            search_opts=search_opts,
            permission_action=None,
            extra_filter=filter,
            **kwargs
        )

        # use explicit scan_params
        search_result = search.params(**scan_params).scan() if scan else search.execute()
```

This way, we can override the params passed to scroll. What do you think?